File: app/models/squirrel.rb
class Squirrel < ActiveRecord::Base
  has_many :branches
  has_many :nuts
  has_many :trees, through: :branches

  def nut_count
    nuts.length
  end

  def tree_count
    trees.length
  end

  def to_hash
    { id:         self.id,
      name:       self.name, 
      tailLength: self.tail_length, 
      treeMail:   self.tree_mail,
      trees:      self.tree_count,
      nuts:       self.nut_count  }
  end

  def self.create_or_update(params)
    squirrel = Squirrel.find_by(id: params[:id])
    if squirrel 
      squirrel.update_attributes(params)
    else
      squirrel = Squirrel.create(params)
    end
    squirrel
  end
end

File: public/js/squirrel.js
function Squirrel(params) {

  this.id = params.id;
  this.name = params.name;
  this.tailLength = params.tailLength;
  this.treeMail = params.treeMail;

  this.nutCount = params.nuts;
  this.treeCount = params.trees;

}

Squirrel.prototype.toJson = function(){
  return JSON.stringify( {name: this.name, tailLength: this.tailLength, treeMail: this.treeMail} );
};
